packages/sdk/src/apiClient/basicAuthProvider.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AxiosRequestConfig } from "axios";
+import { AuthProvider } from "./authProvider";
+
+/**
+ * Provider that handles Basic authentication
+ *
+ * @beta
+ */
+export class BasicAuthProvider implements AuthProvider {
+  private userName: string;
+  private password: string;
+
+  /**
+   *
+   * @param userName - Username used in basic auth
+   * @param password - Password used in basic auth
+   *
+   * @beta
+   */
+  constructor(userName: string, password: string) {
+    this.userName = userName;
```

Comment:
we may need to check whether the value is valid, e.g. non empty

Reply:
Fixed.